fiftyone/core/dataset.py

```diff
@@ -4244,12 +4244,22 @@ def _clear(self, view=None, sample_ids=None):
         else:
             contains_videos = self._contains_videos(any_slice=True)
 
+        ops = []
         if sample_ids is not None:
-            d = {"_id": {"$in": [ObjectId(_id) for _id in sample_ids]}}
+            batch_size = fou.recommend_batch_size_for_value(
+                ObjectId(), max_size=100000
+            )
+
+            for _ids in fou.iter_batches(sample_ids, batch_size):
+                ops.append(
+                    DeleteMany(
+                        {"_id": {"$in": [ObjectId(_id) for _id in _ids]}}
+                    )
+                )
         else:
-            d = {}
+            ops.append(DeleteMany({}))
```

Comment:
Just curious, do we need to explicitly pass in an empty delete here? Does passing in an empty list for ops raise an error?

Reply:
`DeleteMany({})` means delete *everything*, not delete *nothing* 😄